Dockerfile

```diff
@@ -0,0 +1,11 @@
+FROM registry.access.redhat.com/ubi8/nodejs-12:1-98.1634036321 as builder
+
+WORKDIR /opt/app-root/src
+
+COPY package.json .
+COPY package-lock.json .
+RUN npm install
+COPY . .
+RUN npm run build
```

Comment:
this could be reduced just to two steps(image layers)
```Dockerfile
COPY . .
RUN npm install && \
    npm run build
```

Reply:
Most of the time during development workflow and small releases, dependencies do not change. Splitting the Dockerfile takes benefits of this to cache some layers.